### PR TITLE
Internalise localstack setup

### PR DESCRIPTION
**Pull request by cob16**
Including set-up steps for inside docker compose

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -171,13 +178,34 @@ services:
...
+        echo 'run_startup_scripts' >> /usr/local/bin/docker-entrypoint.sh
+
+        bash '/usr/local/bin/docker-entrypoint.sh'
+        echo 'settting up buckets'
+        aws --endpoint-url=http://localhost:4572 s3 mb s3://filestore-bucket
+        aws --endpoint-url=http://localhost:4572 s3api put-bucket-acl --bucket filestore-bucket --acl public-read
+        aws --endpoint-url=http://localhost:4572 s3 mb s3://external-filestore-bucket
+        aws --endpoint-url=http://localhost:4572 s3api put-bucket-acl --bucket external-filestore-bucket --acl public-read
+
+        tail -qF /tmp/localstack_infra.log /tmp/localstack_infra.err
```
Comment (cob16) [inline docker-compose.yml]: We need run our setup after startup but by default after the the script gives up no chance to do so as it tails the logs forever. 

This removes that tail until we are ready

File: docker-compose.yml [hunk 2]
```diff
@@ -159,7 +166,7 @@ services:
       RAILS_ENV: development
       RAILS_LOG_TO_STDOUT: "true"
       REDIS_URL: service-token-cache-redis
-      SERVICE_TOKEN_CACHE_TTL: "300"
+      SERVICE_TOKEN_CACHE_TTL: 999999
```
Comment (cob16) [inline docker-compose.yml]: This needs to really high else the test slug expires

File: docker-compose.yml [hunk 3]
```diff
@@ -151,6 +153,11 @@ services:
 
   service-token-cache-app:
     container_name: service-token-cache-app
+    command:
+      - /bin/sh
+      - -c
+      - | # adds a test token to use in end to end test
+        rails runner "Support::ServiceTokenCache.put('slug', 'token-for-slug')" && bundle exec rails s -e development -p 3000 --binding=0.0.0.0
```
Comment (cob16) [inline docker-compose.yml]: Wondering if adding a test slug should be done in the app instead of here

File: testfile.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+1
```
Comment (asmega) [inline testfile.txt]: don't think you need these anymore?

File: docker-compose.yml (hunk 1 above)
Comment (asmega) [reply docker-compose.yml]: cant say i'm a fan of this. can't we boot everything as normal then call a script to setup for tests?

Comment (cob16) [reply docker-compose.yml]: That on my radar, this way all the docker-compose cmds work as expected. But yeah doing that would be less complex

File: testfile.txt (hunk 4 above)
Comment (cob16) [reply testfile.txt]: opps

File: docker-compose.yml (hunk 1 above)
Comment (cob16) [reply docker-compose.yml]: replaced with a script

**Pull request merged**